README.md

```diff
@@ -2,4 +2,121 @@
+`device`
+`device` specifies the method used for export, and can be either a device or a preflight check. See the Devices section below for possible values. It is a required key.
+
+`encryption_method`:
+`encryption_method` is used exclusively when exporting to USB storage. It is an optional key. Possible values are:
+luks
+
+`encryption_passphrase`
+`encryption_passphrase` is used exclusively when exporting to USB storage. It is an optional key. It contains an arbitrary string that contains the disk encryption passphrase of the device.
+
+
+Example archive metadata (`metadata.json`):
+```
+{
+  "device": "disk"
+  "encryption-method": "luks"
+  "encryption-key": "Your encryption passphrase goes here"
+}
+```
+
+### Devices
+
+For all device types (described in detail below), the following standard error types can be returned:
+
+- `ERROR_FILE_NOT_FOUND`: No file has been specified or the path is incorrect
+- `ERROR_EXTRACTION`: Error while extracting the archive
+- `ERROR_METADATA_PARSING`: The metadata.json file cannot be correctly parsed
+- `ERROR_ARCHIVE_METADATA`: The metadata failed the check
+- `ERROR_USB_CONFIGURATION`: There is no USB controller attached to the VM, the dom0 configuration (in `config.json`) or USB device identifier is is misconfigured
+- `ERROR_GENERIC`: An uncaught (unexpected) error somewhere in the script. These should not happen unless the code improperly handles errors
+
+The supported device types for export are as follows, including the possible errors specific to that device type:
+
+1. `usb-test` : Preflight check that probes for USB connected devices, that returns:
+  - `USB_CONNECTED` if a USB device is attached to the dedicated slot
+  - `USB_NOT_CONNECTED` if no USB is attached
+  - `USB_CHECK_ERROR` if an error occurred during pre-flight
+
+
+2. `disk-test`: Preflight check that checks for LUKS-encrypted volume that returns:
+  - `USB_ENCRYPTED` if a LUKS volume is attached to sd-export
+  - `USB_ENCRYPTION_NOT_SUPPORTED` if a LUKS volume is not attached or there was any other error
+  - `USB_DISK_ERROR`
+
+
+3. `printer-test`: prints a test page that returns:
+  - `ERROR_PRINTER_NOT_FOUND` if no printer is connected
+  - `ERROR_PRINTER_NOT_SUPPORTED` if the printer is not currently supported by the export script
+  - `ERROR_PRINTER_DRIVER_UNAVAILABLE` if the printer driver is not available
+  - `ERROR_PRINTER_INSTALL` If there is an error installing the printer
+  - `ERROR_PRINT` if there is an error printing
+
+
+4. `printer`: sends files to printer that returns:
+  - `ERROR_PRINTER_NOT_FOUND` if no printer is connected
+  - `ERROR_PRINTER_NOT_SUPPORTED` if the printer is not currently supported by the export script
+  - `ERROR_PRINTER_DRIVER_UNAVAILABLE` if the printer driver is not available
+  - `ERROR_PRINTER_INSTALL` If there is an error installing the printer
+  - `ERROR_PRINT` if there is an error printing
+
+
+5. `disk`: sends files to disk:
```

Comment:
it would also be helpful to see the errors that can come up for the export `disk` command, such as: `USB_BAD_PASSPHRASE` and `ERROR_USB_MOUNT`

Reply:
(and `ERROR_USB_WRITE`)

Reply:
great point @creviera, added those error codes and their description